### ISSUE TITLE
E1305 false positive involving dicts

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#287]
Originally reported by: **Chris Rebert (BitBucket: [cvrebert](http://bitbucket.org/cvrebert), GitHub: @cvrebert?)**

---

```
#!bash

$ pylint --version
No config file found, using default configuration
pylint 1.3.0, 
astroid 1.2.0, common 0.62.0
Python 2.7.7 (default, Jun 14 2014, 23:12:13) 
[GCC 4.2.1 Compatible Apple LLVM 5.1 (clang-503.0.40)]
```

```
#!python

# foobarbaz.py
name = 'qwerty'
ret = {'comment': ''}
ret['comment'] = 'MySQL grant {0} is set to be revoked'
ret['comment'] = ret['comment'].format(name)
```

```
#!bash

$ pylint -r n -f parseable foobarbaz.py 
No config file found, using default configuration
************* Module foobarbaz
foobarbaz.py:4: [E1305(too-many-format-args), ] Too many arguments for format string
```

I realize this might be asking too much of the automatic analysis. If pylint could at least instead emit something like a "maybe-too-many-format-args" Warning (rather than Error) in this case, that would be nice.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/287
